docs/rfc/identity_proofs.md

```diff
@@ -0,0 +1,214 @@
+
+The following sections describe how such claims shall be stored in the identity
+payload of a `radicle-link` identity, how to obtain a publishable proof, and how
+to verify it.
+
+## Claims
+
+Claims can only be made by identities of kind `Person`, and claim a single
+external account identifier. They are introduced by defining a new payload type,
+identified by the URL:
+
+    https://radicle.xyz/link/claim/v1
+
+The shape of the JSON structure is:
+
+```json
+{
+    "service": {
+        "name": "STRING",
+        "account": "STRING"
+    },
+    "expiration": {
+        "created": INTEGER,
+        "expires": INTEGER
+    },
+    "proof": "URL"
+}
+```
+
+Where the fields denote:
+
+* `service`
+    * `name`
+
+      A conventional identifier of the external service, e.g. "github",
+      "twitter", "radicle-ethereum"
+
+    * `account`
+
+      The unique account identifier within the service, using the service-
+      specific canonical string representation e.g. "kim",
+      "0x32be343b94f860124dc4fee278fdcbd38c102d88".
+
+* `expiration` (optional)
+    * `created`
+
+      Creation timestamp of the claim, in seconds since 1970-01-01T00:00:00Z.
+
+    * `expires`
+
+      Seconds relative to `created`, after which the claim should no longer be
+      considered.
+
+* `proof` (optional)
+
+  A URL to assist verification tooling in retrieving the proof from the external
+  system. This is mainly a convenience, and obviously requires creation of a new
+  revision after the fact.
+
+## Proof Generation
+
+The above claim payload is committed to the identity history as a new revision.
+Technically, this revision needs to be approved by all key delegations for
+verification to pass later on, but since we assume that eligible keys are held
+by the same person, it may be acceptable to publish the proof right away for
+user experience reasons.
+
+The actual proof consists of the following tuple:
+
+    (root, revision, public key, signature)
+
+> TODO: include a protocol bit (denoting "git"), for forwards compat?
+
+The values `root`, `revision`, and `public key` are specified in
+[identities][ids], and it is RECOMMENDED to follow the serialisation formats
+devised there. `signature` is the Ed25519 signature over `revision`, in much the
+same way as the actual revision is signed. All values can thus be obtained by
+inspecting the identity storage.
+
+It is beyond the scope of this document to devise the exact external format to
+serialise the tuple into, as this is expected to vary from service to service.
+
+## Revocation
+
+A claim can be revoked by creating a new identity revision which simply does not
+contain the claim payload. Likewise, a later claim describing the same `(service
+name, account)` mapping invalidates an earlier one.
+
+> NOTE: claiming multiple accounts per service is permissible in principle, but
+> might be undesirable for certain services. In this case, service-specific
+> post-validation is required.
+
+## Verification
```

Comment:
Should this section also have the step of verifying the claim's `proof`, if present?

Reply:
The proof URL is just a hint where to get the payload from. Say we'd support Twitter, then we can infer https://twitter.com/iamdevloper/status from what we have, but _every_ client would need to scrape the timeline until they find the proof in https://twitter.com/iamdevloper/status/1361327114260271109. Same thing for blockchain proofs, _although_ you could substitute the web3-provider.

Reply:
But what I'm not understanding is where this comes into play as part of the attestation. Is it up to the malkovich to check this or up to the client program? Is that why there's no recommendation on what to do with it?

Reply:
There isn't really anything to check. Say all you have is the identity doc. Now you invoke `rad sozial überprüfung`, and that reads the payload, sees aha deep-link into twitter, let's go and get the 4-tuple from there. If you already have the 4-tuple, you can omit this step. If you _don't_ have a 4-tuple, _nor_ a deeplink, that nifty program needs to traverse the twitter timeline until it finds a tweet which _looks_ like it could contain a 4-tuple.

The drawback is of course that adding this URL creates a new revision.